common/workunit/workunit.cpp

```diff
@@ -11408,9 +11426,28 @@ extern WORKUNIT_API void submitWorkUnit(const char *wuid, const char *username,
     Owned<IWorkUnit> workunit = factory->updateWorkUnit(wuid);
     assertex(workunit);
 
-    SCMStringBuffer token;
-    createToken(wuid, username, password, token);
-    workunit->setSecurityToken(token.str());
+    //Digitally sign the workunit
+    IDigitalSignatureManager * pDSM = createDigitalSignatureManagerInstanceFromEnv();
```

Comment:
Can we move all of this signature code into a common method.  Perhaps replace 
  workunit->setSecuritySignature(const char *);

With something like:
  bool workunit->signWorkunit();
or
  bool workunit->generateSecuritySignature();
??

Then these sections become:
if (!workunit->sign();)
    {
        if (isEmptyString(password))
            throw makeStringException(0, "Unable to sign workunit");
  }